packages/coderskit/src/components/Steps/Steps.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { HTMLAttributes, ReactNode, ReactElement, LabelHTMLAttributes } from 'react';
+import classnames from 'classnames';
+import styled from '@emotion/styled';
+import { ThemeColorsKeys } from '../..';
+
+// SINGLE STEP
+
+export type StepState = 'success' | 'error' | 'active' | 'pending';
+
+export interface StepProps extends HTMLAttributes<HTMLDivElement> {
+  state?: StepState;
+  labelLayout?: LabelLayout;
+  children?: ReactNode;
+}
+
+interface StepContentProps extends StepProps {
+  size?: number;
+  color?: ThemeColorsKeys;
+  fontColor?: ThemeColorsKeys;
+  children?: ReactNode;
+}
+
+export const StepContentBase = styled.div<StepContentProps>(props => {
+  const { colors } = props.theme;
+  const color = colors[props.color!];
+  const fontColor = colors[props.fontColor!];
+
+  return {
+    display: 'flex',
+    borderRadius: '100%',
+    alignItems: 'center',
+    justifyContent: 'center',
+    border: `1px solid ${color}`,
+    backgroundColor: color,
+    color: fontColor,
+    width: props.size,
+    height: props.size,
+  };
+});
+
+const StepContentSuccess = styled(StepContentBase)(props => {
+  const { colors } = props.theme;
+
+  return {
+    backgroundColor: colors.primary,
+    borderColor: colors.primary,
+    color: colors.white,
+  };
+});
+
+const StepContentError = styled(StepContentBase)(props => {
+  const { colors } = props.theme;
+
+  return {
+    backgroundColor: colors.error,
+    borderColor: colors.error,
+    color: colors.white,
+  };
+});
+
+const StepContentActive = styled(StepContentBase)(props => {
+  const { colors } = props.theme;
+
+  return {
+    backgroundColor: colors.white,
+    borderColor: colors.primary,
+    color: colors.primary,
+  };
+});
+
+const StepContentPending = styled(StepContentBase)(props => {
+  const { colors } = props.theme;
+
+  return {
+    backgroundColor: colors.white,
+    borderColor: colors.fontDisabled,
+    color: colors.fontDisabled,
+  };
+});
+
+const StepVertical = styled.div<StepProps>(() => {
+  return {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    margin: '10px',
+  };
+});
+
+const StepHorizontal = styled.div<StepProps>(() => {
+  return {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    margin: '10px',
+
+    '.ck-step-content': {
+      marginRight: 8,
+    },
+  };
+});
+
+export const StepContent = (props: StepContentProps) => {
+  const StepContent =
```

Comment:
Tę logikę lepiej wyrzuć do jakiejś funkcji, to łatwiej będzie to czytać. Najlepiej zrobić switch